pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -534,6 +534,18 @@ func (s *DiscoveryServer) edsUpdate(shard, serviceName string,
 	defer s.mutex.Unlock()
 	requireFull := false
 
+	// To prevent memory leak.
+	// Should delete the service EndpointShards, when endpoints deleted or service deleted.
+	if len(istioEndpoints) == 0 {
+		if s.EndpointShardsByService[serviceName] != nil {
+			delete(s.EndpointShardsByService[serviceName].Shards, shard)
+			if len(s.EndpointShardsByService[serviceName].Shards) == 0 {
+				delete(s.EndpointShardsByService, serviceName)
```

Comment:
Now endPoint deletion not trigger ConfigUpdate?

Reply:
We donot need this to trigger ConfigUpdate, because service deletion will do

Reply:
@hzxuzhonghu 
ConfigUpdate does not need to triggered when the endpoint change from 1 to 0 ?